document/JS.bs

```diff
@@ -0,0 +1,921 @@
+    1. Let |n| be |descriptor|.initial.
+    1. If |descriptor|.maximum is present, let |m| be |descriptor|.maximum; otherwise, let |m| be empty.
+    1. If |m| is not empty and |m| &lt; |n|, throw a {{RangeError}} exception.
+    1. Let |type| be the [=table type=] {𝗆𝗂𝗇 n, 𝗆𝖺𝗑 |m|} 𝖺𝗇𝗒𝖿𝗎𝗇𝖼.
+    1. Let |store| be the current agent cluster's [=associated store=].
+    1. Let (|store|, |tableaddr|) be [=alloc_table=](|store|, |type|). <!-- TODO(littledan): Report allocation failure https://github.com/WebAssembly/spec/issues/584 -->
+    1. Set the current agent cluster's [=associated store=] to |store|.
+    1. [=Create a table object=] from the table instance |tableaddr| and return the result.
+</div>
+
+<div algorithm>
+    The <dfn for="Table">grow(d)</dfn> method, when invoked, performs the following steps:
+    1. Let |tableaddr| be the table instance's \[[Table]] internal slot.
+    1. Let |initialSize| be the length of the table instance's \[[Values]] internal slot.
+    1. Let |store| be the current agent cluster's [=associated store=].
+    1. Let |result| be [=grow_table=](|store|, |tableaddr|, |d|).
+    1. If |result| is [=error=], throw a [=RangeError=] exception.
+
+    Note: The above exception may happen due to either insufficient memory or an invalid size parameter.
+
+    6. Set the current agent cluster's [=associated store=] to |store|.
+    1. [=Append=] null to the table instance's \[[Values]] internal slot |d| &minus; |initialSize| times.
+    1. Return |initialSize|.
+</div>
+
+<div algorithm>
+  The getter of the <dfn for="Table">length</dfn> attribute of {{Table}} returns the length of the table's \[[Values]] internal slot.
+</div>
+
+<div algorithm>
+    The <dfn for="Table">get(index)</dfn> method, when invoked, performs the following steps:
+    1. Let |values| be the table instance's \[[Values]] internal slot.
+    1. Let |size| be the length of |values|.
+    1. If |index| &ge; |size|, throw a [=RangeError=] exception.
+    1. Return |values|[|index|].
+</div>
+
+<div algorithm>
+    The <dfn for="Table">set(index, value)</dfn> method, when invoked, performs the following steps:
+    1. Let |tableaddr| be the table instance's \[[Table]] internal slot.
+    1. Let |values| be the table instance's \[[Values]] internal slot.
+    1. If |value| is null, let |elem| be an empty [=function element=].
+    1. Otherwise, let |funcaddr| be |value|.\[[FunctionAddress]].
+    1. Let |store| be the current agent cluster's [=associated store=].
+    1. Let |store| be [=write_table=](|store|, |tableaddr|, |index|, |funcaddr|).
+    1. If |store| is [=error=], throw a [=RangeError=] exception.
+    1. Set the current agent cluster's [=associated store=] to |store|.
+    1. Set |values|[|index|] to |value|.
+    1. Return undefined.
+</div>
+
+<h3 id="exported-function-exotic-objects">Exported Functions</h3>
+
+<pre class='idl'>
+[NoInterfaceObject]
+interface ExportedFunction : Function { };
```

Comment:
Are these Web IDL semantics actually what's intended? We define a new class, not exposed to the global, as if via the following?

```js
class ExportedFunction extends Function {
  constructor() {
    throw new TypeError("Illegal invocation");
  }
}
```

And the APIs accepting import/export records will throw if they do not pass the brand check for `ExportedFunction`?

I think this probably shouldn't be a Web IDL type, and you should just use `Function` in the typedefs for import/export records.

Reply:
You're right, that's not intended. Changing it to Function and adding the appropriate TypeErrors/checks explicitly. (Actually there was a typo--for imports, we allow ordinary functions, not just Exported Functions).